src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,119 @@
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private User testUser;
+
+    @BeforeEach
+    void beforeEach() {
+        testUser = createValidUser();
+    }
+
+    @AfterEach
+    void afterEach() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_userUnderMinAge_notOk() {
+        testUser.setAge(USER_MIN_AGE - 1);
+        InvalidUserException exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(testUser));
+        assertEquals(exception.getMessage(), "Users under "
+                + USER_MIN_AGE + " are not allowed!");
+    }
+
+    @Test
+    void register_userWithNullAge_notOk() {
+        User user = createValidUser();
+        user.setAge(null);
+        InvalidUserException exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(exception.getMessage(), "Users under "
+                + USER_MIN_AGE + " are not allowed!");
+    }
+
+    @Test
+    void register_userWithShortLogin_notOk() {
+        User user = createValidUser();
+        user.setLogin("short");
+        InvalidUserException exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(exception.getMessage(), "Login must be at least "
+                + USER_LOGIN_PASSWORD_MIN_LENGTH + " characters");
+    }
+
+    @Test
+    void register_userWithNullLogin_notOk() {
+        User user = createValidUser();
+        user.setLogin(null);
+        InvalidUserException exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(exception.getMessage(), "Login must be at least "
+                + USER_LOGIN_PASSWORD_MIN_LENGTH + " characters");
+    }
+
+    @Test
+    void register_userWithShortPassword_notOk() {
+        User user = createValidUser();
+        user.setPassword("pass");
+        InvalidUserException exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(exception.getMessage(), "Password must be at least "
+                + USER_LOGIN_PASSWORD_MIN_LENGTH + " characters");
+    }
+
+    @Test
+    void register_userWithNullPassword_notOk() {
+        User user = createValidUser();
+        user.setPassword(null);
+        InvalidUserException exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(exception.getMessage(), "Password must be at least "
+                + USER_LOGIN_PASSWORD_MIN_LENGTH + " characters");
+    }
+
+    @Test
+    void register_userWithExistingLogin_notOk() {
+        User user1 = createValidUser();
+        User user2 = createValidUser();
+        registrationService.register(user1);
```

Comment:
we should not put existing user via `service.register()` method, because this method is under the test right now, and we shouldn't use it to arrange the tests data, use low-level approach with `storageDao` or `Storage` instead
check common mistakes